[splash-screen] Add configuration guide for dark-mode (#8288)
expo/expo: packages/expo-splash-screen/README.md

Selecting this resize mode requires some more work to be done in native configuration.
Please take a look at the [`res/drawable/splashscreen.xml`](#resdrawablesplashscreenxml) and [`res/drawable/splashscreen_background.png`](#resdrawablesplashscreen_backgroundpng) sections.

### Per-appearance (a.k.a. dark-mode) splash screen

> `expo-splash-screen` supports per-appearance splash screen that responds to system appearance changes on iOS 13+ and dark-mode changes on Android 10+.

Reviewer: ```suggestion
`expo-splash-screen` supports per-appearance splash screens that respond to system appearance changes on iOS 13+ and dark-mode changes on Android 10+.
```

> The guide below shows how to configure your Xcode project to use single image file as a splash screen using a `.storyboard` file (configuration for `.xib` filetype is analogous).

Reviewer: ```suggestion
The guide below shows how to configure your Xcode project to use a single image file as a splash screen using a `.storyboard` file (configuration for `.xib` filetype is analogous).
```

1. [Add an image to `Images.xcassets`](#-add-an-image-to-imagesxcassets)
2. [Create `SplashScreen.storyboard`](#-create-splashscreenstoryboard)

> 3. [Select `Content Mode` for the `ImageView` in `SplashScreen.sptryboard`](#-select-content-mode-for-the-imageview-in-splashscreenstoryboard)

Reviewer: ```suggestion
3. [Select `Content Mode` for the `ImageView` in `SplashScreen.storyboard`](#-select-content-mode-for-the-imageview-in-splashscreenstoryboard)
```

3. Locate `App Icons and Launch Images` section and `Launch Screen File` option.
4. Select or enter `SplashScreen` as the value for located option.

<details>
 <summary>Show image with details</summary>
<img src="./assets/configuration-ios-selectLaunchScreen.png" height="350" />
</details>

#### 🛠 (<em>optional</em>) Enable dark mode

##### Provide different background colors

> Depending on what iOS version your application is targeting you have to adjust your native project differently to obtain working per-appearance splash screen view.

Reviewer: ```suggestion
Depending on what iOS version your application is targeting, you have to adjust your native project differently to obtain a working per-appearance splash screen view.
```

##### I'm targeting iOS 11+

> You can take advantage of [`named colors`](https://developer.apple.com/library/archive/documentation/DeveloperTools/Conceptual/WhatsNewXcode/xcode_9/xcode_9.html) feature in your Xcode project.

Reviewer: ```suggestion
You can take advantage of [`named colors`](https://developer.apple.com/library/archive/documentation/DeveloperTools/Conceptual/WhatsNewXcode/xcode_9/xcode_9.html) in your Xcode project.
```

> 1. Create new `Color Set` and customize it's values for different color modes:

Reviewer: ```suggestion
1. Create a new `Color Set` and customize its values for different color modes:
```

>     - in your `.xcassets` directory (create new `.xcassets` for colors or reuse existing one e.g. with images) create `New Color Set` and name it `SplashScreenBackground`,

Reviewer: ```suggestion
    - in your `.xcassets` directory (either create a new `.xcassets` for colors, or reuse an existing one e.g. with images) create `New Color Set` and name it `SplashScreenBackground`,
```

    - navigate to `Attributes Inspector` in the right panel and change `Appearance` to `Any, Dark`,
    - select desired color in `Attributes Inspector` in the right panel for each mode.

<details>
 <summary>Show image with details</summary>
<img src="./assets/configuration-ios-createNamedColor.png" height="350" />
</details>

2. Select created `named color` as the `Background` for the `Image View` in `SplashScreen.storyboard`:
    - open `SplashScreen.storyboard` and select `Image View` in view hierarchy inspector,
    - navigate to `Attributes Inspector` in the right panel,

>     - configure `Background` parameter selecting created `named color` (that should be listed as `SplashScreenBackground`).

Reviewer: ```suggestion
    - configure `Background` parameter by selecting your created `named color` (that should be listed as `SplashScreenBackground`).
```

<details>
 <summary>Show image with details</summary>
<img src="./assets/configuration-ios-selectNamedColor.png" height="350" />
</details>

##### I'm targeting iOS version < 11

You cannot use [`named colors`](https://developer.apple.com/library/archive/documentation/DeveloperTools/Conceptual/WhatsNewXcode/xcode_9/xcode_9.html) feature in your Xcode project.

> Instead you have to create additional image set that would contain small 1x1px images, each with desired background color. Then you'll use this additional image resource as a background in splash screen view.

Reviewer: ```suggestion
Instead you have to create an additional image set that contains small 1x1px images, each with the desired background color. Then, you'll use this additional image resource as a background in the splash screen view.
```

> You can use this online online generator to obtain 1x1px `.png` images with desired colors: http://www.1x1px.me.

Reviewer: ```suggestion
You can use this online generator to obtain 1x1px `.png` images with desired colors: http://www.1x1px.me.
```

1. Create `SplashScreenBackground` `Image Set` with desired background colors for each mode in your `Images.xcassets` directory:
    - open your `.xcassets` directory with images,
    - in the content panel add `New image set` and name it `SplashScreenBackground`,
    - convert this `Image set` to support `Dark Appearance` by navigating to `Attributes Inspector` in the right panel and changing `Appearance` to `Any, Dark`,
    - provide images with colors for every mode (you can generate color 1x1px images using http://www.1x1px.me).

<details>
 <summary>Show image with details</summary>
<img src="./assets/configuration-ios-addSplashScreenBackgroundImages.png" height="250" />
</details>

> 2. Update `SplashScreen.storyboard` to consist of single top-level `View` with to `Image View` subviews (solid-colored image in the background and actual splash screen image in the foreground):

Reviewer: ```suggestion
2. Update `SplashScreen.storyboard` to consist of a single top-level `View` with two `Image View` subviews (solid-colored image in the background and actual splash screen image in the foreground):
```

    - open `SplashScreen.storyboard` and replace `Image View` with a plain `View` (search `Library` for it and drag&drop it in place of current `Image View`),
    - add two 

3. Configure first `Image View` (background color):
    - configure attributes in `Attributes Inspector`:
        - set `Image` to `SplashScreenBackground` (prepared in previous step),
        - set `Content Mode` to `Scale To Fill` (color needs to take all available space),
    - make this subview take all available space in parent view:
        - open `Add new constraints` bottom menu,
        - make sure `Constrain to margin` is not checked,

>         - for every input open dropdown and select `View` (parent view reference) and set `0` as the value,

Reviewer: ```suggestion
        - for every input, open the dropdown and select `View` (parent view reference) and set `0` as the value,
```

        - once every side is covered (`0` value and parent view reference selected in dropdown) hit `Add 4 Constraints`,
        - observe that in `View Hierarchy Inspector` constraints are added and `Image View` resized to take whole place of parent view.

<details>
 <summary>Show image with details</summary>
<img src="./assets/configuration-ios-createBackgroundImageView.png" height="350" />
</details>

4. Configure second `Image View` (actual splash screen image):
    - select second `Image View` and select correct `Image` in `Attributes Inspector` alongside with desired `Content Mode`,
    - make this subview take all available space in parent view (see previous step).
     
<details>
 <summary>Show image with details</summary>
<img src="./assets/configuration-ios-addConstraintsToImageViews.png" height="350" />
</details>

##### Provide different splash screen image

> You might want to add separate image for `dark` mode. If the system is switched to `dark` mode it would pick this different image instead of normal one and present it in splash screen view.

Reviewer: ```suggestion
You might want to add a separate image for `dark` mode. If the system is switched to `dark` mode, it would pick this different image instead of the normal one and present it in the splash screen view.
```

> 1. In your Xcode project open `SplashScreen` created in previous section.

Reviewer: ```suggestion
1. In your Xcode project open `SplashScreen` (created in previous section).
```

2. Convert this asset to support `Dark Appearance`:
  - navigate to `Attributes Inspector` in the right panel,
  - locate `Appearances` section and select `Any, Dark`,
  - provide image for `dark mode` by dropping it to the correct box.

<details>
 <summary>Show image with details</summary>
<img src="./assets/configuration-ios-addDarkImagesXcassets.png" height="280" />
</details>

###### Background color when you want to support iOS < 11

> If you're targeting version of iOS < 11 then you cannot use `named color` feature and instead you need to generate images with desired background colors that are going to be used as the background for splash screen view.

Reviewer: ```suggestion
If you're targeting a version of iOS < 11 then you cannot use `named color` feature and instead you need to generate images with desired background colors that are going to be used as the background for splash screen view.
```

</manifest>
```

#### 🛠 (<em>optional</em>) Enable dark mode

##### Provide different background colors - `res/values-night/colors.png`

> If you want to have different background color in splash screen depending on system color mode, you need to create similar file to `colors.xml`, but in directory `res/values-night`.

Reviewer: ```suggestion
If you want to have different background colors in your splash screen depending on the system color mode, you need to create a similar file to `colors.xml`, but in the directory `res/values-night`.
```

Values in this file are going to picked by the system when it is switched to `dark` mode.

```diff
<resources>
+ <color name="splashscreen_background">#AABBCC</color> <!-- #AARRGGBB or #RRGGBB format -->
</resources>
```

##### Provide different splash screen image - `res/drawable-night/splashscreen_image.png`

> You might want to provide separate splash screen image for dark mode usage and place it under the `res/drawable-night` directory with exactly same name as a normal one.

Reviewer: ```suggestion
You might want to provide a separate splash screen image for dark mode usage, and place it under the `res/drawable-night` directory with exactly the same name as your normal one.
```
